bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/builder/ChannelBuilder.java

```diff
@@ -140,10 +140,6 @@ public ChannelBuilder withDefaultTags(Set<String> defaultTags) {
      * @return channel builder
      */
     public ChannelBuilder withKind(ChannelKind kind) {
-        if (kind == null) {
```

Comment:
I am wondering if there was a reason why @sjka has not removed this check after he added the `@NonNullbyDefault` annotation to the class. I have removed the check for now, but if there was a good reason to keep it we should think about allowing `@Nullable` for the kind and keep the check.

Reply:
I don't know what has been @sjka intentions and I agree that this could be removed...
... but didn't we have a discussion in another topic where you / Kai argued that we cannot rely on correct usage WRT to the nullness annotations for code that is called by third party ones (e.g. binding developers etc.)?
So, if we don't trust non ESH framework developers that he uses annotations classes correctly, isn't that exactly what you want?

Reply:
I am still struggling with which parts of the code are internal for the framework and which is used by others. I mean `public` doesn't mean its really exported by the manifest, so I would have to look it up to make that decision since I don't know the code by heart as you guys :)

So if this method is really "public public", I think we should add a `@Nullable` on the parameter and keep the check.